src/contract/deploy.rs

```diff
@@ -46,17 +48,26 @@ impl<T: Transport> Builder<T> {
     pub fn execute<P, V>(self, code: V, params: P, from: Address) -> Result<PendingContract<T>, ethabi::Error>
     where
         P: Tokenize,
-        V: Into<Vec<u8>>,
+        V: AsRef<str>,
     {
         let options = self.options;
         let eth = self.eth;
         let abi = self.abi;
 
+        let mut code_hex = code.as_ref().to_string();
+
+        for (lib, address) in self.linker {
+            let replace = format!("__{:_<1$}", lib, 38-lib.len());
+            code_hex = code_hex.replacen(&replace, &address.to_hex(), 1);
+        }
+        code_hex = code_hex.replace("\"", "").replace("0x", ""); // This is to fix truffle + serde_json redundant `"` and `0x`
+        let code = code_hex.into_bytes();
```

Comment:
That conversion is not valid, we need to convert `str` into `bytes`, but `str` should be hex-encoded, so we need to use `from_hex()` here.

Previously that function accepted raw bytes, so `from_hex()` was happening earlier.

Reply:
Wow, I don't know how I missed that, you're right, before it wasn't hex and it shouldn't be hex now.